src/com/palmergames/bukkit/towny/tasks/HourlyTimerTask.java

```diff
@@ -56,5 +60,24 @@ private void decrementJailedHours() {
 					resident.setJailHours(resident.getJailHours() - 1);
 					resident.save();
 				}
+		if (TownySettings.hourlyJailFee() >= 0)
+		{
+			// Check if towns can afford upkeep for prisoners and if not to release them
+			for (Resident resident : new ArrayList<>(universe.getJailedResidentMap()))
+			{
+				Town town = resident.getJailTown();
+				double hourlyJailFee = TownySettings.hourlyJailFee();
+				if (!town.getAccount().canPayFromHoldings(hourlyJailFee))
+				{
+					Bukkit.getScheduler().runTaskLater(plugin, () -> JailUtil.unJailResident(resident, UnJailReason.INSUFFICIENT_FUNDS), 20);
+				}
+				else
+				{
+					TownyMessaging.sendPrefixedTownMessage(town, Translatable.of("msg_x_has_been_withdrawn_for_upkeep_of_prisoner_x", hourlyJailFee,resident));
+					town.getAccount().withdraw(hourlyJailFee, "Jailee Upkeep");
```

Comment:
Adding the resident name to the "Jailee upkeep for " + resident.getName() will make this clearer in the bankhistory books.

Reply:
Change made to to include resident name for bankhistory books.